exp/views/study.py

```diff
@@ -445,43 +468,59 @@ def manage_researcher_permissions(self):
         Handles adding, updating, and deleting researcher from study. Users are
         added to study read group by default.
         """
+        change_requester = self.request.user
         study = self.get_object()
-        study_read_group = study.study_read_group
-        study_admin_group = study.study_admin_group
-        add_user = self.request.POST.get("add_user")
-        remove_user = self.request.POST.get("remove_user")
-        update_user = None
+        study_researcher_group = study.researcher_group
+        study_admin_group = study.admin_group
+        id_of_user_to_add = self.request.POST.get("add_user")
+        id_of_user_to_remove = self.request.POST.get("remove_user")
 
         # Early exit if the user doesn't have proper permissions.
-        if not self.request.user.groups.filter(name=study_admin_group.name).exists():
-            messages.error(
-                self.request,
-                f"You don't have proper permissions to add researchers to {study.name}.",
-            )
-            return
+        # We should not need this because the test function will check for us whether
+        # or not this user is legal for the POST request based on object perms.
+        # if not self.request.user.groups.filter(name=study_admin_group.name).exists():
+        #     messages.error(
+        #         self.request,
+        #         f"You don't have proper permissions to add researchers to {study.name}.",
+        #     )
+        #     return
 
         if self.request.POST.get("name") == "update_user":
-            update_user = self.request.POST.get("pk")
-            permissions = self.request.POST.get("value")
-        if add_user:
+            id_of_user_to_update = self.request.POST.get("pk")
+            name_of_role_to_enable = self.request.POST.get("value")
+            if id_of_user_to_update:
+                user_to_update = User.objects.get(pk=id_of_user_to_update)
+                if name_of_role_to_enable == "study_admin":
+                    # if admin, removes user from study read and adds to study admin
+                    user_to_update.groups.add(study_admin_group)
+                    user_to_update.groups.remove(study_researcher_group)
+                    self.send_study_email(user_to_update, "admin")
+                if name_of_role_to_enable == "study_read":
+                    # if read, removes user from study admin and adds to study read
+                    # Must have more than one admin to make this change.
+                    if study_admin_group.user_set.count() > 1:
+                        user_to_update.groups.add(study_researcher_group)
+                        user_to_update.groups.remove(study_admin_group)
+                        self.send_study_email(user_to_update, "read")
```

Comment:
does this need to change now to encompass all the new study roles?

Reply:
Yep - we've got new roles and new permissions. Wherever we're using old roles and perms, we need to change.